### PR TITLE
feat(manage-ant-modal): initial adding and opening logic for manage a…

### PR DESCRIPTION
**Pull request by atticusofsparta**
…nt modal with ant state mapping

### CODE REVIEW

File: src/components/layout/tables/AntRow/AntRow.tsx [hunk 1]
```diff
@@ -75,6 +81,13 @@ function AntRow({
               }
             : {}
         }
+        onClick={
+          isMobile
+            ? () => setShowManageModal(true)
+            : () => {
+                return;
+              }
```
Comment (dtfiedler) [inline src/components/layout/tables/AntRow/AntRow.tsx:89]: not showing modal on mobile?

File: src/components/layout/tables/RowItem/RowItem.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+          style={textColor ? { color: textColor, flex: 3 } : { flex: 3 }}
+        >
+          {details[1]}
+        </td>
+
+        <td
+          className="assets-table-item flex-right"
+          style={textColor ? { color: textColor } : {}}
+        >
+          {details[2] ? details[2] : <></>}
```
Comment (dtfiedler) [inline src/components/layout/tables/RowItem/RowItem.tsx:40]: nit: can we make this dynamic

File: src/components/layout/tables/AntRow/AntRow.tsx (hunk 1 above)
Comment (atticusofsparta) [reply src/components/layout/tables/AntRow/AntRow.tsx:89]: backwards actually, if its mobile the row itself is the button that pops the modal, if not mobile it is not a button and the button to pop the modal is the manage button from the manage assets buttons

File: src/components/layout/tables/RowItem/RowItem.tsx (hunk 2 above)
Comment (atticusofsparta) [reply src/components/layout/tables/RowItem/RowItem.tsx:40]: maybe. Need to look at it more.

File: src/components/modals/ManageAntModal/ManageAntModal.tsx [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+                key={`${contractId}-confirmations`}
+                confirmations={confirmations}
+              />,
+            ]}
+            bgColor={'#1E1E1E'}
+            textColor={'var(--text-white)'}
+          />
+          <RowItem
+            details={[
+              `Associated Names: ${associatedNames.length}`,
```
Comment (dtfiedler) [inline src/components/modals/ManageAntModal/ManageAntModal.tsx:116]: nit: the design does not include the count. if we are going to include it I would put it as

`Associated Names (COUNT): `,

image

File: src/components/layout/tables/AntRow/AntRow.tsx (hunk 1 above)
Comment (dtfiedler) [reply src/components/layout/tables/AntRow/AntRow.tsx:89]: ah misread - some small items on mobile look crunched

image

File: src/components/modals/ManageAntModal/ManageAntModal.tsx [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+              <button
+                className="button"
+                key={`${contractId}-nickname-edit-button`}
+              >
+                <PencilIcon width={20} height={20} fill="var(--text-white)" />
+              </button>,
+            ]}
+            bgColor={'#1E1E1E'}
+            textColor={'var(--text-white)'}
+          />
```
Comment (dtfiedler) [inline src/components/modals/ManageAntModal/ManageAntModal.tsx:147]: nit: can we crate an object and use it to dynamically create these `RowItems`


e.g.
```
const antDetails = {
    'Nickname':  state?.name ? state.name : 'N/A',
    'Ticker': state?.ticker ? state.ticker : 'N/A',
    'Target ID': targetId ? targetId : 'N/A',
    ....etc.
}
```

and then use it like
```
Object.entries(antDetails).map(([key, value], index) => (
    <RowItem
            details={[
              key,
              value
              <button
                className="button"
                key={`${contractId}-${key}-edit-button`}
              >
                <PencilIcon width={20} height={20} fill="var(--text-white)" />
              </button>,
            ]}
            bgColor={'#1E1E1E'}
            textColor={'var(--text-white)'}
          />
)
```

File: src/components/layout/tables/AntRow/AntRow.tsx (hunk 1 above)
Comment (dtfiedler) [reply src/components/layout/tables/AntRow/AntRow.tsx:89]: i also feel like a magnifying glass might be better for viewing the ant details on mobile than clicking the status icon. it's unclear that that will show this page ☝️ - what do you think?

File: src/components/modals/ManageAntModal/ManageAntModal.tsx (hunk 4 above)
Comment (dtfiedler) [reply src/components/modals/ManageAntModal/ManageAntModal.tsx:147]: for reference - we are doing something similar to ☝️  in the `AntCard` as it displays similar details
https://github.com/ar-io/arns-react/blob/develop/src/components/cards/AntCard/AntCard.tsx#L99-L108

**Comment by dtfiedler** [comment]
looking good! a few minor comments to make things a bit more dynamic

**Review by dtfiedler — CHANGES_REQUESTED** [review]
a few minor comments to make things a bit more dynamic, otherwise - looking good!

File: src/components/layout/tables/AntRow/AntRow.tsx (hunk 1 above)
Comment (atticusofsparta) [reply src/components/layout/tables/AntRow/AntRow.tsx:89]: Ok yeah I need to take a look at some text alignment it seems. Hm, I dislike using the search icon for something that isnt running a search, what about this? 
image

File: src/components/modals/ManageAntModal/ManageAntModal.tsx (hunk 3 above)
Comment (atticusofsparta) [reply src/components/modals/ManageAntModal/ManageAntModal.tsx:116]: Right - need to discuss this at the design meet tomorrow. An ant can have many names associated with it, and they can take up more space than is available - so I added `count` to display the total number. Still waiting on alot of my questions to be answered, but IK lucas has been working on it today so hopefully ill get one soon.

File: src/components/layout/tables/AntRow/AntRow.tsx (hunk 1 above)
Comment (atticusofsparta) [reply src/components/layout/tables/AntRow/AntRow.tsx:89]: or maybe this (standardish fullscreen icon) 
image

File: src/components/modals/ManageAntModal/ManageAntModal.tsx (hunk 4 above)
Comment (atticusofsparta) [reply src/components/modals/ManageAntModal/ManageAntModal.tsx]: updated

File: src/components/modals/ManageAntModal/ManageAntModal.tsx [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+  }
+
+  function getAssociatedNames() {
+    const domains: string[] = [];
+    Object.entries(arnsSourceContract.records).map(([name, id]) => {
+      if (id === contractId.toString()) {
+        domains.push(name);
+      }
+    });
+    return domains;
```
Comment (dtfiedler) [inline src/components/modals/ManageAntModal/ManageAntModal.tsx:52]:
```suggestion
  function getAssociatedNames(txId: ArweaveTransactionID) {
    return Object.entries(arnsSourceContract.records).map(([name, id]) => {
      if (id === txId.toString()) return name;
    });
  }
```

File: src/components/modals/ManageAntModal/ManageAntModal.tsx [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+  const [{ arnsSourceContract }] = useGlobalState();
+  const modalRef = useRef(null);
+  const [associatedNames, setAssociatedNames] = useState<string[]>([]);
+  const [antDetails, setAntDetails] = useState<any[]>();
+  const isMobile = useIsMobile();
+  // todo: manage asset modal writes asset modifications to contract. It will auto detect if the asset is an ANT, name, or undername.
+  // if the asset is a name, it will write modifications to the registry. If its an undername, it will write mods to the ant. If its an ant, it will write mods to the ant.
+
+  useEffect(() => {
+    setAssociatedNames(getAssociatedNames());
```
Comment (dtfiedler) [inline src/components/modals/ManageAntModal/ManageAntModal.tsx:34]:
```suggestion
```

File: src/components/modals/ManageAntModal/ManageAntModal.tsx [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+    Object.entries(arnsSourceContract.records).map(([name, id]) => {
+      if (id === contractId.toString()) {
+        domains.push(name);
+      }
+    });
+    return domains;
+  }
+
+  function setDetails() {
+    const details = [
```
Comment (dtfiedler) [inline src/components/modals/ManageAntModal/ManageAntModal.tsx:56]: const associatedNames = getAssociatedNames(contractId);

Comment (dtfiedler) [inline src/components/modals/ManageAntModal/ManageAntModal.tsx:56]:
```suggestion
  function setDetails() {
    const associatedNames = getAssociatedNames(contractId);
    const details = [
```

File: src/components/modals/ManageAntModal/ManageAntModal.tsx [hunk 8]
```diff
@@ -0,0 +1,277 @@
...
+}: {
+  contractId: ArweaveTransactionID;
+  setShowModal: (show: boolean) => void;
+  state: ANTContractState;
+  confirmations: number;
+  targetId?: string;
+}) {
+  const [{ arnsSourceContract }] = useGlobalState();
+  const modalRef = useRef(null);
+  const [associatedNames, setAssociatedNames] = useState<string[]>([]);
```
Comment (dtfiedler) [inline src/components/modals/ManageAntModal/ManageAntModal.tsx:27]:
```suggestion
```

File: src/components/modals/ManageAntModal/ManageAntModal.tsx [hunk 9]
```diff
@@ -0,0 +1,277 @@
...
+                  copyText={contractId.toString()}
+                  size={24}
+                  wrapperStyle={{
+                    display: 'flex',
+                    justifyContent: 'center',
+                    alignItems: 'center',
+                    textColor: 'var(--bright-white)',
+                  }}
+                />
+              )}
```
Comment (dtfiedler) [inline src/components/modals/ManageAntModal/ManageAntModal.tsx:258]: it's more concise to do this in the copy text component
```suggestion
              {isMobile ? (
                <CopyTextButton
                  displayText={isMobile ? `${contractId
                    .toString()
                    .slice(0, 10)}...${contractId.toString().slice(isMobile ? -10): contractId.toString()}`}
                  copyText={contractId.toString()}
                  size={24}
                  wrapperStyle={{
                    display: 'flex',
                    justifyContent: 'center',
                    alignItems: 'center',
                    textColor: 'var(--bright-white)',
                  }}
                />
              ) : (
                <CopyTextButton
                  displayText={contractId.toString()}
                  copyText={contractId.toString()}
                  size={24}
                  wrapperStyle={{
                    display: 'flex',
                    justifyContent: 'center',
                    alignItems: 'center',
                    textColor: 'var(--bright-white)',
                  }}
                />
              )}
```

File: src/components/modals/ManageAntModal/ManageAntModal.tsx (hunk 5 above)
Comment (atticusofsparta) [reply src/components/modals/ManageAntModal/ManageAntModal.tsx:52]: added

File: src/components/modals/ManageAntModal/ManageAntModal.tsx (hunk 9 above)
Comment (atticusofsparta) [reply src/components/modals/ManageAntModal/ManageAntModal.tsx:258]: added

File: src/components/modals/ManageAntModal/ManageAntModal.tsx (hunk 8 above)
Comment (atticusofsparta) [reply src/components/modals/ManageAntModal/ManageAntModal.tsx:27]: removed

File: src/components/modals/ManageAntModal/ManageAntModal.tsx (hunk 7 above)
Comment (atticusofsparta) [reply src/components/modals/ManageAntModal/ManageAntModal.tsx]: added

File: src/components/modals/ManageAntModal/ManageAntModal.tsx (hunk 6 above)
Comment (atticusofsparta) [reply src/components/modals/ManageAntModal/ManageAntModal.tsx]: removed

File: src/components/modals/ManageAntModal/ManageAntModal.tsx [hunk 10]
```diff
@@ -0,0 +1,264 @@
...
+                  <span
+                    className="assets-manage-button"
+                    key={`${contractId}-name`}
+                  >
+                    {name}
+                  </span>
+                  &nbsp;
+                </>
+              ))
+            : 'N/A'}
```
Comment (dtfiedler) [inline src/components/modals/ManageAntModal/ManageAntModal.tsx:90]: image
for now, can we do just a joined list of names, vs. the buttons
```suggestion
        <td className="assets-table-item" key={`${contractId}-data`}>
          {`Associated Names (${names?.length}) :`}
        </td>,
        <td
          className="assets-table-item"
          style={{ flex: 4 }}
          key={`${contractId}-data`}
        >
          {names.length ? names.join(', ') :  'N/A'}
```

**Review by dtfiedler — APPROVED** [review]

**Pull request merged**